File: Gelman_BDA_ARM/arm/3.5.py
import numpy
from matplotlib import pyplot as plt
import statsmodels.api as sm
import sim

ks = numpy.loadtxt("kidscore.dat",  skiprows=1)
olsmod = sm.OLS(ks[:,1], sm.add_constant(ks[:,3], prepend=True))
olsres = olsmod.fit()
print olsres.summary()
plt.plot (ks[:,3], ks[:,1], '.')

# simulate data points
beta, sigma = sim.sim_lm(olsres)

# plot simulation

x = numpy.linspace(60,150,100)

for i in range(10): 
    plt.plot(x, beta[i,0] + beta[i,1]*x, 'y')

plt.plot(x, olsres.params[0] + olsres.params[1]*x, 'b')
plt.xlabel("Mother IQ Score")
plt.ylabel("Child Test Score")
        
plt.show()

File: Gelman_BDA_ARM/arm/3.5.R
library ("foreign")
library("arm")

iq.data <- read.dta ("~/kod/mcmc/doc/gelman/ARM_Data/child.iq/kidiq.dta")
attach(iq.data)

fit.2 <- lm (kid_score ~ mom_iq)
fit.2.sim <- sim (fit.2)
plot (mom_iq, kid_score, xlab="Mother IQ score", ylab="Child test score")

print (fit.2.sim)

for (i in 1:10){
  curve (fit.2.sim$coef[i,1] + fit.2.sim$coef[i,2]*x, add=TRUE,col="gray")
}
curve (coef(fit.2)[1] + coef(fit.2)[2]*x, add=TRUE, col="black")
